### PR TITLE
ICU-22402 Add options to PersonNameFormatter and update nativeSpaceRe…

### PR DESCRIPTION
**Pull request by richgillam**
…placement logic

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22402
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by pedberg-icu — APPROVED** [review]

File: icu4j/main/classes/core/src/com/ibm/icu/impl/personname/PersonNameFormatterImpl.java [hunk 1]
```diff
@@ -227,7 +225,16 @@ private String[] asStringArray(ICUResourceBundle resource) {
...
+        // if the formatter has its display order set to one of the "force" values, that overrides
+        // all this logic and the name's preferred-order property
+        if (this.displayOrder == PersonNameFormatter.DisplayOrder.FORCE_GIVEN_FIRST) {
+            return true;
+        } else if (this.displayOrder == PersonNameFormatter.DisplayOrder.FORCE_SURNAME_FIRST) {
+            return false;
+        }
+
         // the name can declare its order-- check that first (it overrides any locale-based calculation)
+        // [unless the formatter's display order is FORCE_NATIVE_ORDERING, which is handled below]
```
Comment (richgillam) [inline icu4j/main/classes/core/src/com/ibm/icu/impl/personname/PersonNameFormatterImpl.java:237]: Whoops.  Need to take this out...

File: icu4j/main/classes/core/src/com/ibm/icu/impl/personname/PersonNameFormatterImpl.java [hunk 2]
```diff
@@ -237,6 +244,8 @@ private boolean nameIsGnFirst(PersonName name) {
         // Otherwise, search the gnFirstLocales and snFirstLocales for the locale's name.
         // For our purposes, the "locale's name" is the locale the name itself gives us (if it
         // has one), or the locale we guess for the name (if it doesn't).
+        // If the formatter's display order is FORCE_NATIVE_ORDERING, use the formatter's locale
+        // here instead of the name's locale.
```
Comment (richgillam) [inline icu4j/main/classes/core/src/com/ibm/icu/impl/personname/PersonNameFormatterImpl.java:248]: Whoops.  Need to take this out too...

**Review by pedberg-icu — DISMISSED** [review]

**Comment by richgillam** [comment]
Holding until API change is approved...

**Comment by richgillam** [comment]
(The forced push was me adding `@stable` directives in the Javadoc for the new enum constants and squashing my commits again, both in one step.)

**Comment by richgillam** [comment]
The API change was approved at this morning's ICU TC meeting.

**Review by pedberg-icu — APPROVED** [review]

**Pull request merged**